compiler/rustc_session/src/config.rs

```diff
@@ -1312,12 +1318,13 @@ pub enum CrateType {
     Staticlib,
     Cdylib,
     ProcMacro,
+    Sdylib,
 }
 
 impl CrateType {
     pub fn has_metadata(self) -> bool {
         match self {
-            CrateType::Rlib | CrateType::Dylib | CrateType::ProcMacro => true,
+            CrateType::Rlib | CrateType::Dylib | CrateType::ProcMacro | CrateType::Sdylib => true,
```

Comment:
Same comment as in https://github.com/rust-lang/rust/pull/134767/files#r2001606923.